plugins/gstperf.c

```diff
@@ -401,16 +422,91 @@ gst_perf_cpu_get_load (GstPerf * perf, guint32 * cpu_load)
   } else {
     *cpu_load = 0;
   }
+
   /*Remember the total and idle CPU for the next check */
   perf->prev_cpu_total = total;
   perf->prev_cpu_idle = idle;
+
+  return TRUE;
+
+cpu_failed:
+  GST_ERROR_OBJECT (perf, "Failed to get the CPU load");
+  return FALSE;
+}
+
+#elif IS_MACOSX
+static gboolean
+gst_perf_cpu_get_load_macosx (GstPerf * perf, guint32 * cpu_load)
+{
+  guint32 idle = 0;
+  guint32 total = 0;
+  guint32 diff_total, diff_idle;
+  host_cpu_load_info_data_t cpuinfo;
+  mach_msg_type_number_t count = HOST_CPU_LOAD_INFO_COUNT;
+
+  g_return_val_if_fail (perf, FALSE);
+  g_return_val_if_fail (cpu_load, FALSE);
+
+  /* Default value in case of failure */
+  *cpu_load = -1;
+
+  if (host_statistics (mach_host_self (), HOST_CPU_LOAD_INFO,
+          (host_info_t) & cpuinfo, &count) == KERN_SUCCESS) {
+    for (int i = 0; i < CPU_STATE_MAX; i++) {
+      total += cpuinfo.cpu_ticks[i];
```

Comment:
does  `cpu_ticks` have the same length (`CPU_STATE_MAX`)?  and all its values are valid for the calculation?

Reply:
Yes, according to the docs:
```c
struct host_cpu_load_info {                 /* number of ticks while running... */
    unsigned long cpu_ticks[CPU_STATE_MAX]; /* ... in the given mode */
};
```

I've initialized the array so that all fields are zero.